[16.0][ADD] helpdesk_mgmt_account: links account moves and helpdesk tickets.

nicolas-delbovier-acsone opened the pull request:
<img width="1496" height="679" alt="image" src="https://github.com/user-attachments/assets/b30f0c53-ec65-43c6-84ca-c4aadd4aa8ed" />

<img width="1496" height="679" alt="image" src="https://github.com/user-attachments/assets/d14ab9dc-a01f-4e0b-875d-d29834fda820" />

nicolas-delbovier-acsone committed:
[ADD] helpdesk_mgmt_account: links account moves and helpdesk tickets.

nicolas-delbovier-acsone committed:
[ADD] helpdesk_mgmt_account: links account moves and helpdesk tickets.

marcelsavegnago commented:
@nicolas-delbovier-acsone please make sure to fill in the README.rst files with the module info or remove any empty ones.

lmignon commented on helpdesk_mgmt_account/models/account_move.py:
```diff
@@ -0,0 +1,19 @@
+    @api.depends("ticket_ids")
+    def _compute_ticket_count(self):
+        for move in self:
+            move.ticket_count = len(move.ticket_ids)
```
@nicolas-delbovier-acsone you should use a read_groupto avoid perf issue if the field is accessed on a large recordsed....

lmignon replied:
ping @nicolas-delbovier-acsone

nicolas-delbovier-acsone committed:
[FIX] helpdesk_mgmt_account: remove incorrect Sale Order context
The button to view Helpdesk tickets from an Account Move was incorrectly
passing a 'default_sale_order_id' in the action context due to a
copy-paste error from the sales module.

This caused new tickets created from an Account Move view to be linked to an
unrelated Sale Order.

nicolas-delbovier-acsone committed:
[FIX] helpdesk_mgmt_account: remove incorrect Sale Order context
The button to view Helpdesk tickets from an Account Move was incorrectly
passing a 'default_sale_order_id' in the action context due to a
copy-paste error from the sales module.

This caused new tickets created from an Account Move view to be linked to an
unrelated Sale Order.

nicolas-delbovier-acsone committed:
[REF] helpdesk_mgmt_account: move open tickets action to Python
Shifting the action logic to Python facilitates dynamic customization. This approach allows inheriting modules to override or extend the action via super(), avoiding the limitations and complexity of XML xpath inheritance for conditional logic

nicolas-delbovier-acsone committed:
[REF] helpdesk_mgmt_account: move open tickets action to Python
Shifting the action logic to Python facilitates dynamic customization. This approach allows inheriting modules to override or extend the action via super(), avoiding the limitations and complexity of XML xpath inheritance for conditional logic

nicolas-delbovier-acsone committed:
[IMP] helpdesk_mgmt_account: enhance action context validation in tests
Following the removal of the erroneous 'default_sale_order_id' from
the account move action, this commit strengthens the test suite to
prevent regressions.

The test logic now explicitly validates the content of the action
context. This ensures that:
- The expected Account Move ID is present.
- No unrelated context keys (like 'default_sale_order_id') are
  leaking into the view from inherited methods or copy-paste errors.

nicolas-delbovier-acsone committed:
[IMP] helpdesk_mgmt_account: enhance action context validation in tests
Following the removal of the erroneous 'default_sale_order_id' from
the account move action, this commit strengthens the test suite to
prevent regressions.

The test logic now explicitly validates the content of the action
context. This ensures that:
- The expected Account Move ID is present.
- No unrelated context keys (like 'default_sale_order_id') are
  leaking into the view from inherited methods or copy-paste errors.

nicolas-delbovier-acsone committed:
[IMP] helpdesk_mgmt_account: complete description and configuration readme

nicolas-delbovier-acsone committed:
[IMP] helpdesk_mgmt_account: complete description and configuration readme

nicolas-delbovier-acsone committed:
[IMP] helpdesk_mgmt_account: remove empty readme files

nicolas-delbovier-acsone committed:
[PERF] helpdesk_mgmt_account: use read_group to reduce db calls

nicolas-delbovier-acsone committed:
[PERF] helpdesk_mgmt_account: use read_group to reduce db calls

marcelsavegnago commented:
@nicolas-delbovier-acsone could you fix pre-commit ?

nicolas-delbovier-acsone commented:
> @nicolas-delbovier-acsone could you fix pre-commit ?

There you go !

marcelsavegnago reviewed (approved):
LGTM

marcelsavegnago commented:
/ocabot merge nobump

marcelsavegnago commented:
/ocabot merge nobump

pedrobaeza commented:
/ocabot merge nobump
